chemprop/nn/metrics.py

```diff
@@ -63,7 +63,9 @@ def forward(
         lt_mask: Tensor,
         gt_mask: Tensor,
     ):
-        return self._calc_unreduced_loss(preds, targets, mask, lt_mask, gt_mask)[mask].mean()
+        return self._calc_unreduced_loss(preds.unsqueeze(-1), targets, mask, lt_mask, gt_mask)[
```

Comment:
Although the shape of the model output changes from $(b, (t * s))$ to $(b, t, s)$. As you use `preds = preds[..., 0]` in `_evaluate_batch` in `MPNN()`, the shape should be $(b, t)$, which is still the same shape as the targets.

Reply:
What you said is true. However, the same `self._calc_unreduced_loss` is shared for both the loss function and the metrics. For example, the RMSE metrics and the RMSE loss. When it's used in the loss function, the preds fed in is $(b,t,1)$. When it's used by the metrics, the preds fed in is $(b,t)$. To make it work for both cases, the `self._calc_unreduced_loss` expects preds to be $(b,t,1)$, so anything fed in from the metrics end need to be unsqueezed.

Reply:
It would be a bit strange to me to use unsqueeze and squeeze for the metrics calculations. Perhaps, it would be better not to use `preds = preds[..., 0]` in `_evaluate_batch` and instead chunk the first elements in the last dimension for loss functions and metrics that only need the prediction values.